upgrade cockroachdb/datadriven to v1.0.2 to remove archived dependencies

pacoxu opened the pull request:
Signed-off-by: Paco Xu <paco.xu@daocloud.io>


Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

I found that after https://github.com/etcd-io/raft/pull/20 was merged, etcd v3.5.x was not updated.

So I sync the dependencies update to `release-3.5` branch.

ahrtr commented:
Please make sure your local go version is 1.17.13, then execute `./scripts/fix.sh`

pacoxu commented:
> Please make sure your local go version is 1.17.13, then execute `./scripts/fix.sh`

Updated:
```
docker run --rm -it -v "$PWD":/usr/src/etcd -w /usr/src/etcd m.daocloud.io/docker.io/library/golang:1.17.13-bullseye bash
./scripts/fix.sh
```
I run the script in a golang:1.17.13 container and updated the PR.
